source/NVDAObjects/IAccessible/mozilla.py

```diff
@@ -16,11 +16,33 @@
 from logHandler import log
 import textInfos.offsets
 from NVDAObjects.behaviors import RowWithFakeNavigation
+from virtualBuffers import VirtualBuffer
+import api
 from . import IA2TextTextInfo
 from . import ia2Web
 
 class Mozilla(ia2Web.Ia2Web):
 
+	def _getPhysicalFocus(self):
+		try:
+			paccFocus=self.IAccessibleObject.accParent.accFocus
+		except COMError:
+			return
+		if paccFocus:
+			return IAccessible(IAccessibleObject=IAccessibleHandler.normalizeIAccessible(paccFocus),IAccessibleChildID=0)
+
+	def setFocus(self):
+		oldFocus=self._getPhysicalFocus()
+		super(Mozilla,self).setFocus()
+		# Although all versions of Firefox block in setFocus until the physical focus has moved,

```

Comment:
This isn't quite correct. Firefox 57+ setFocus may return before the physical focus has moved. However, a subsequent call to accFocus will definitely block until the physical focus has moved. Same result, so no code change needed, but probably worth clarifying the comment.